src/common/components/table/table-examples/TxTableCellRenderers.tsx

```diff
@@ -106,24 +112,63 @@ export const TxLinkCellRenderer = (value: string) => {
   );
 };
 
-export const AddressLinkCellRenderer = (value: string) => {
-  return (
-    <AddressLink principal={value} variant="tableLink">
+export const AddressLinkCellRenderer = (value: TxTableAddressColumnData) => {
+  const { address, isContract } = value;
+  return address && isContract ? (
+    <Flex
+      gap={1}
+      alignItems="center"
+      bg="surfacePrimary"
+      borderRadius="md"
+      py={0.5}
+      px={1}
+      w="fit-content"
+      _groupHover={{
+        bg: 'surfaceTertiary',
+      }}
+    >
+      <Icon h={3} w={3} color="iconSecondary">
+        <ClarityIcon />
+      </Icon>
+      <AddressLink principal={address} variant="tableLink">
+        <Text
+          whiteSpace="nowrap"
+          overflow="hidden"
+          textOverflow="ellipsis"
+          textStyle="text-regular-xs"
+          color="textSecondary"
+          _hover={{
+            color: 'textInteractiveHover',
+          }}
+        >
+          {getContractName(address)}
+        </Text>
+      </AddressLink>
+    </Flex>
+  ) : address && !isContract ? (
+    <AddressLink principal={address} variant="tableLink">
       <Text whiteSpace="nowrap" overflow="hidden" textOverflow="ellipsis" fontSize="sm">
-        {value}
+        {truncateStxAddress(address)}
       </Text>
     </AddressLink>
+  ) : (
+    <Text whiteSpace="nowrap" overflow="hidden" textOverflow="ellipsis" fontSize="sm">
```

Comment:
This component is used many times, consider creating a component EllipsisText and reuse it.